File: java-todo-app/src/test/java/com/example/todo/domain/TodoRepositoryTest.java
package com.example.todo.domain;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;

import static org.assertj.core.api.Assertions.assertThat;

@DataJpaTest
class TodoRepositoryTest {

    @Autowired
    private TodoRepository todoRepository;

    @Test
    public void save_and_get_todo() {
        Todo todoToSave = Todo.of("어떤 할 일");
        Todo savedTodo = todoRepository.save(todoToSave);

        assertThat(savedTodo).isEqualTo(todoToSave);

        Todo fetchedTodo = todoRepository.findById(savedTodo.getId()).orElseThrow();

        assertThat(savedTodo).isEqualTo(fetchedTodo);
    }

    @Test
    public void save_and_delete_todo() {
        Todo todoToSave = Todo.of("어떤 할 일");
        Todo savedTodo = todoRepository.save(todoToSave);
        Long id = savedTodo.getId();

        assertThat(todoRepository.existsById(id)).isTrue();

        todoRepository.deleteById(savedTodo.getId());

        assertThat(todoRepository.existsById(id)).isFalse();
    }
}

File: kotlin-todo-app/src/test/kotlin/com/example/todo/domain/TodoRepositoryTest.kt
package com.example.todo.domain

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest

@DataJpaTest
class TodoRepositoryTest {

    @Autowired
    private lateinit var todoRepository: TodoRepository

    @Test
    fun `Todo는 영속 및 조회가 가능하다`() {
        val todoToSave = Todo.of("할 일")
        val savedTodo = todoRepository.save(todoToSave)

        assertThat(savedTodo).isEqualTo(todoToSave)

        val fetchedTodo = todoRepository.findById(savedTodo.id).orElseThrow()

        assertThat(fetchedTodo).isEqualTo(savedTodo)
    }

    @Test
    fun `영속된 Todo는 삭제할 수 있다`() {
        val savedTodo = todoRepository.save(Todo.of("할 일"))

        assertThat(todoRepository.existsById(savedTodo.id)).isTrue()

        todoRepository.deleteById(savedTodo.id)

        assertThat(todoRepository.existsById(savedTodo.id)).isFalse()
    }
}
